charts/cluster-addons/templates/monitoring/loki-stack.yaml

```
@@ -1,86 +1,101 @@
-{{- if and .Values.monitoring.enabled .Values.monitoring.lokiStack.enabled }}
+{{- if and .Values.monitoring.enabled .Values.monitoring.loki.enabled }}
 ---
 apiVersion: v1
 kind: Secret
 metadata:
-  name: {{ include "cluster-addons.componentName" (list . "loki-stack") }}-config
+  name: {{ include "cluster-addons.componentName" (list . "loki") }}-config
   labels:
-    {{- include "cluster-addons.componentLabels" (list . "loki-stack") | nindent 4 }}
+    {{- include "cluster-addons.componentLabels" (list . "loki") | nindent 4 }}
     addons.stackhpc.com/watch: ""
 stringData:
   defaults: |
     loki:
       serviceMonitor:
         enabled: true
-    promtail:
+      monitoring:
+        dashboards:
+          enabled: false
+
+  overrides: |
+    {{- toYaml .Values.monitoring.loki.release.values | nindent 4 }}
+---
+apiVersion: v1
+kind: Secret
+metadata:
+  name: {{ include "cluster-addons.componentName" (list . "promtail") }}-config
+  labels:
+    {{- include "cluster-addons.componentLabels" (list . "promtail") | nindent 4 }}
+    addons.stackhpc.com/watch: ""
+stringData:
+  defaults: |
+    config:
       serviceMonitor:
         enabled: true
-      # Get promtail to scrape systemd services
-      config:
-        snippets:
-          extraScrapeConfigs: |
-            - job_name: journal
-              journal:
-                path: /var/log/journal
-                max_age: 12h
-                labels:
-                  job: systemd-journal
-              relabel_configs:
-                - source_labels: ['__journal__systemd_unit']
-                  target_label: 'unit'
-                - source_labels: ['__journal__hostname']
-                  target_label: 'hostname'
-                - source_labels: ['__journal_priority_keyword']
-                  target_label: level
-      # Mount journal directory and machine-id file into promtail pods
-      extraVolumes:
-        - name: journal
-          hostPath:
-            path: /var/log/journal
-        - name: machine-id
-          hostPath:
-            path: /etc/machine-id
-      extraVolumeMounts:
-        - name: journal
-          mountPath: /var/log/journal
-          readOnly: true
-        - name: machine-id
-          mountPath: /etc/machine-id
-          readOnly: true
-    grafana:
-      sidecar:
-        datasources:
-          enabled: false
+      clients:
+        {{- if not (and .Values.monitoring.promtail.release.values.config (and .Values.monitoring.promtail.release.values.config.clients (ne (index .Values.monitoring.promtail.release.values.config.clients 0).url ""))) }}
+        - url: http://loki-stack.{{ .Values.monitoring.loki.release.namespace }}.svc:3100/loki/api/v1/push
+          tenant_id: 1
+        {{- end }}
```

Why do we need this conditional? I think we can probably just drop it and have
```suggestion
        - url: http://loki-stack.{{ .Values.monitoring.loki.release.namespace }}.svc:3100/loki/api/v1/push
          tenant_id: 1
```
If a user of these charts wants to add their own list of clients they are free to overwrite it completely using `.Values.monitoring.loki.release.values`. I think trying to second-guess the user and add client URLs conditionally is more likely to just cause confusion.